packages/react-components/src/components/ErrorBar.tsx

```diff
@@ -123,29 +128,115 @@ export interface ActiveError {
  * Active errors are selected by {@link ErrorBarProps.activeErrors}.
  *
+ * This component internally tracks dismissed by the user.
+ *   * Errors that have an associated timestamp: The error is shown on the UI again if it occurs after being dismissed.
+ *   * Errors that do not have a timestamp: The error is dismissed until it dissappears from the props.
+ *         If the error recurs, it is shown in the UI.
+ *
  * Uses {@link @fluentui/react#MessageBar} UI element.
  */
 export const ErrorBar = (props: ErrorBarProps): JSX.Element => {
   const localeStrings = useLocale().strings.errorBar;
   const strings = props.strings ?? localeStrings;
 
-  if (props.activeErrors.length === 0) {
-    return <></>;
-  }
+  const [dismissedErrors, setDismissedErrors] = useState<DismissedError[]>([]);
+
+  // dropDismissalsForInactiveErrors only returns a new object if `dismissedErrors` actually changes.
+  // Without this behaviour, this `useEffect` block would cause a render loop.
+  useEffect(
+    () => setDismissedErrors(dropDismissalsForInactiveErrors(props.activeErrors, dismissedErrors)),
+    [props.activeErrors, dismissedErrors]
+  );
+
+  const toShow = errorsToShow(props.activeErrors, dismissedErrors);
 
-  // FIXME: Memoize onDismiss callbacks.
   return (
     <Stack>
-      {props.activeErrors.map((activeError) => (
+      {toShow.map((activeError) => (
         <MessageBar
           {...props}
           key={activeError.type}
           messageBarType={MessageBarType.error}
-          onDismiss={() => props.onDismissErrors([activeError.type])}
+          onDismiss={() => setDismissedErrors(dismissError(dismissedErrors, activeError))}
         >
           {strings[activeError.type]}
         </MessageBar>
       ))}
     </Stack>
   );
 };
+
+interface DismissedError {
+  type: ErrorType;
+  dismissedAt: Date;
+  activeSince?: Date;
+}
+
+// Always returns a new Array so that the state variable is updated, trigerring a render.
+const dismissError = (dismissedErrors: DismissedError[], toDismiss: ActiveError): DismissedError[] => {
+  const now = new Date(Date.now());
+  for (const error of dismissedErrors) {
+    if (error.type === toDismiss.type) {
+      // Bump the timestamp for latest dismissal of this error to now.
+      error.dismissedAt = now;
+      error.activeSince = toDismiss.timestamp;
+      return Array.from(dismissedErrors);
+    }
+  }
+
+  // Record that this error was dismissed for the first time right now.
+  return [
+    ...dismissedErrors,
+    {
+      type: toDismiss.type,
+      dismissedAt: now,
+      activeSince: toDismiss.timestamp
+    }
+  ];
+};
+
+// Returns a new Array iff contents change, to avoid re-rendering when nothing was dropped.
+const dropDismissalsForInactiveErrors = (
+  activeErrors: ActiveError[],
+  dismissedErrors: DismissedError[]
+): DismissedError[] => {
+  const active = new Map();
+  for (const error of activeErrors) {
+    active[error.type] = error;
```

Comment:
use a plain object or change this line to 
active.set(error.type, error);

This mapObj[index] won't visit it through map, it will add a property into the obj instead 
Check "Setting object properties" in page https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Map for reference